tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_gemv_kk.hpp

```diff
@@ -0,0 +1,228 @@
+
+  // make alpha and beta consistent with types in KokkosBals::gemv
+  const auto alpha = static_cast<typename decltype(A_view)::value_type>(1);
+  const auto beta  = static_cast<typename decltype(y_view)::value_type>(0);
+  KokkosBlas::gemv("N", alpha, A_view, x_view, beta, y_view);
+}
+
+//
+// updating gemv: z = y + Ax
+//
+// for now, specialize for default_accessor
+// https://github.com/kokkos/stdBLAS/issues/122
+//
+template<class ExeSpace,
+         class ElementType_A,
+         std::experimental::extents<>::size_type numRows_A,
+         std::experimental::extents<>::size_type numCols_A,
+         class Layout_A,
+         class ElementType_x,
+         std::experimental::extents<>::size_type ext_x,
+         class Layout_x,
+         class ElementType_y,
+         std::experimental::extents<>::size_type ext_y,
+	 class Layout_y,
+         class ElementType_z,
+         std::experimental::extents<>::size_type ext_z,
+         class Layout_z>
+void matrix_vector_product(kokkos_exec<ExeSpace> /*kexe*/,
+			   std::experimental::mdspan<
+			     ElementType_A,
+			     std::experimental::extents<numRows_A, numCols_A>,
+			     Layout_A,
+			     std::experimental::default_accessor<ElementType_A>
+			   > A,
+			   std::experimental::mdspan<
+			     ElementType_x,
+			     std::experimental::extents<ext_x>,
+			     Layout_x,
+			     std::experimental::default_accessor<ElementType_x>
+			   > x,
+			   std::experimental::mdspan<
+			     ElementType_y,
+			     std::experimental::extents<ext_y>,
+			     Layout_y,
+			     std::experimental::default_accessor<ElementType_y>
+			   > y,
+			   std::experimental::mdspan<
+			     ElementType_z,
+			     std::experimental::extents<ext_z>,
+			     Layout_z,
+			     std::experimental::default_accessor<ElementType_z>
+			   > z)
+{
+
+  // constraints
+  static_assert(A.rank() == 2);
+  static_assert(x.rank() == 1);
+  static_assert(y.rank() == 1);
+  static_assert(z.rank() == 1);
+
+  // preconditions
+  if ( A.extent(1) != x.extent(0) ){
+    throw std::runtime_error("KokkosBlas: matrix_vector_product: A.extent(1) != x.extent(0) ");
+  }
+  if ( A.extent(0) != y.extent(0) ){
+    throw std::runtime_error("KokkosBlas: matrix_vector_product: A.extent(0) != y.extent(0) ");
+  }
+  if ( A.extent(0) != z.extent(0) ){
+    throw std::runtime_error("KokkosBlas: matrix_vector_product: A.extent(0) != z.extent(0) ");
+  }
+
+  // this print is detected in the tests
+#if defined KOKKOS_STDBLAS_ENABLE_TESTS
+  std::cout << "updating_matrix_vector_product: kokkos impl\n";
+#endif
+
+  auto A_view = Impl::mdspan_to_view(A);
+  auto x_view = Impl::mdspan_to_view(x);
+  auto y_view = Impl::mdspan_to_view(y);
+  auto z_view = Impl::mdspan_to_view(z);
+
+  // FRIZZI: we need to improve this or maybe we should
+  // just do two ops: fist y->z, and then gemv directly
+  auto ex = ExeSpace();
+  Kokkos::parallel_for(Kokkos::RangePolicy(ex, 0, z_view.extent(0)),
+		       KOKKOS_LAMBDA (const std::size_t & i)
```

Comment:
Would you consider passing this index by value instead of by const reference?  On computers with 64-bit pointers, there's no benefit to using a const reference here.